File: app/src/main/java/ru/magflayer/spectrum/data/database/dao/NcsColorDao.java
package ru.magflayer.spectrum.data.database.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import ru.magflayer.spectrum.data.entity.NcsColor;

import static android.arch.persistence.room.OnConflictStrategy.REPLACE;

@Dao
public interface NcsColorDao {

    @Query("SELECT * FROM NcsColor")
    List<NcsColor> loadNcsColors();

    @Query("SELECT * FROM NcsColor WHERE hex=:hex")
    NcsColor loadNcsColorByHex(String hex);

    @Query("SELECT COUNT(*) FROM NcsColor")
    int getRowCount();

    @Insert(onConflict = REPLACE)
    long[] saveNcsColors(NcsColor... ncsColors);

}

File: app/src/main/java/ru/magflayer/spectrum/data/database/dao/NcsColorDao.kt
package ru.magflayer.spectrum.data.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import ru.magflayer.spectrum.data.entity.NcsColor

@Dao
interface NcsColorDao {

    @get:Query("SELECT COUNT(*) FROM NcsColor")
    val rowCount: Int

    @Query("SELECT * FROM NcsColor")
    suspend fun loadNcsColors(): List<NcsColor>

    @Query("SELECT * FROM NcsColor WHERE hex=:hex")
    suspend fun loadNcsColorByHex(hex: String): NcsColor

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun saveNcsColors(vararg ncsColors: NcsColor): LongArray
}
